tests/net/context/src/main.c

```diff
@@ -998,43 +1036,44 @@ static void test_init(void)
 
 	/* The semaphore is there to wait the data to be received. */
 	k_sem_init(&wait_data, 0, UINT_MAX);
+
+	return NULL;
+}
+
+static void before(void *dummy)
+{
+	ARG_UNUSED(dummy);
+	net_ctx_create();
+	net_ctx_bind_fail();
+	net_ctx_bind_uni_success_v6();
+	net_ctx_bind_uni_success_v4();
+	net_ctx_bind_mcast_success();
+	net_ctx_listen_v6();
+	net_ctx_listen_v4();
+	net_ctx_connect_v6();
+	net_ctx_connect_v4();
+	net_ctx_accept_v6();
+	net_ctx_accept_v4();
 }
 
-/*test case main entry*/
-void test_main(void)
+static void teardown(void *dummy)
 {
-	ztest_test_suite(test_context,
-			 ztest_unit_test(test_init),
-			 ztest_unit_test(test_net_ctx_get_fail),
-			 ztest_unit_test(test_net_ctx_get_all),
-			 ztest_unit_test(test_net_ctx_get_success),
-			 ztest_unit_test(test_net_ctx_create),
-			 ztest_unit_test(test_net_ctx_bind_fail),
-			 ztest_unit_test(test_net_ctx_bind_uni_success_v6),
-			 ztest_unit_test(test_net_ctx_bind_uni_success_v4),
-			 ztest_unit_test(test_net_ctx_bind_mcast_success),
-			 ztest_unit_test(test_net_ctx_listen_v6),
-			 ztest_unit_test(test_net_ctx_listen_v4),
-			 ztest_unit_test(test_net_ctx_connect_v6),
-			 ztest_unit_test(test_net_ctx_connect_v4),
-			 ztest_unit_test(test_net_ctx_accept_v6),
-			 ztest_unit_test(test_net_ctx_accept_v4),
-			 ztest_unit_test(test_net_ctx_send_v6),
-			 ztest_unit_test(test_net_ctx_send_v4),
-			 ztest_unit_test(test_net_ctx_sendto_v6),
-			 ztest_unit_test(test_net_ctx_sendto_v4),
-			 ztest_unit_test(test_net_ctx_recv_v6),
-			 ztest_unit_test(test_net_ctx_recv_v4),
-			 ztest_unit_test(test_net_ctx_recv_v6_fail),
-			 ztest_unit_test(test_net_ctx_recv_v4_fail),
-			 ztest_unit_test(test_net_ctx_recv_v6_again),
-			 ztest_unit_test(test_net_ctx_recv_v4_again),
-			 ztest_unit_test(test_net_ctx_recv_v6_reconfig),
-			 ztest_unit_test(test_net_ctx_recv_v4_reconfig),
-			 ztest_unit_test(test_net_ctx_recv_v6_timeout),
-			 ztest_unit_test(test_net_ctx_recv_v4_timeout),
-			 ztest_unit_test(test_net_ctx_recv_v6_timeout_forever),
-			 ztest_unit_test(test_net_ctx_recv_v4_timeout_forever),
-			 ztest_unit_test(test_net_ctx_put));
-	ztest_run_test_suite(test_context);
+	struct net_if *iface;
+
+	iface = net_if_get_first_by_type(&NET_L2_GET_NAME(DUMMY));
+	zassert_not_null(iface, "Interface is NULL");
+
+	zassert_true(net_if_ipv6_addr_rm(iface, &in6addr_my),
+		     "net_if_ipv6_addr_rm fail");
+
+	zassert_true(net_if_ipv4_addr_rm(iface, &in4addr_my),
+		     "net_if_ipv4_addr_rm fail");
+
+	net_ipv6_addr_create(&in6addr_mcast, 0xff02, 0, 0, 0, 0, 0, 0, 0x0001);
+
+	zassert_true(net_if_ipv6_maddr_rm(iface, &in6addr_mcast),
+		     "net_if_ipv6_maddr_rm mcast fail");
 }
+
+ZTEST_SUITE(net_context_get, NULL, setup, NULL, NULL, teardown);
```

Comment:
Let's rename `setup` and `teardown` to something a bit more specific since they're weak symbols that could be overridden by some other unrelated function named `setup` or `teardown`

Reply:
OK, updated